shared/common-adapters/profile-card.tsx

```diff
@@ -0,0 +1,243 @@
+import FloatingMenu from './floating-menu'
+import Icon from './icon'
+import Meta from './meta'
+import ProgressIndicator from './progress-indicator'
+import Text from './text'
+import WithTooltip from './with-tooltip'
+import FollowButton from '../profile/user/actions/follow-button'
+
+const Kb = {
+  Box,
+  Box2,
+  ClickableBox,
+  ConnectedNameWithIcon,
+  FloatingMenu,
+  Icon,
+  Meta,
+  ProgressIndicator,
+  Text,
+  WithTooltip,
+}
+
+type Props = {
+  clickToProfile?: true
+  showClose?: true
+  username: string
+}
+
+const maxIcons = 4
+
+type ServiceIconsProps = {
+  userDetails: Tracker2Types.Details
+}
+
+const assertionTypeToServiceId = (assertionType): Platforms.ServiceId | null => {
+  switch (assertionType) {
+    case 'facebook':
+    case 'github':
+    case 'hackernews':
+    case 'keybase:':
+    case 'reddit:':
+    case 'twitter':
+      return assertionType
+    default:
+      return null
+  }
+}
+
+const ServiceIcons = ({userDetails}: ServiceIconsProps) => {
+  const services = new Map(
+    userDetails.assertions
+      ? userDetails.assertions.toArray().map(([_, assertion]) => [assertion.type, assertion.value])
+      : []
+  )
+  const serviceIds = [...services]
+    .map(([serviceName]) => assertionTypeToServiceId(serviceName))
+    .filter(Boolean) as Array<Platforms.ServiceId>
+  const [expanded, setExpanded] = React.useState(false)
+  const expandLabel =
+    !expanded && serviceIds.length > maxIcons ? `+${serviceIds.length - (maxIcons - 1)}` : ''
+  return (
+    <Kb.Box2
+      direction="horizontal"
+      gap="xtiny"
+      style={styles.serviceIcons}
+      fullWidth={true}
+      centerChildren={true}
+    >
+      {serviceIds.slice(0, expandLabel ? maxIcons - 1 : undefined).map(serviceId => (
+        <Kb.WithTooltip
+          key={serviceId}
+          tooltip={`${services.get(serviceId)} on ${capitalize(serviceId)}`}
+          position="top center"
+        >
+          <Kb.Icon fontSize={14} type={Platforms.serviceIdToIcon(serviceId)} />
+        </Kb.WithTooltip>
+      ))}
+      {!expanded && serviceIds.length > 4 && (
+        <Kb.ClickableBox onClick={() => setExpanded(true)} style={styles.expand}>
+          <Kb.Meta title={`+${serviceIds.length - 3}`} backgroundColor={Styles.globalColors.greyDark} />
+        </Kb.ClickableBox>
+      )}
+    </Kb.Box2>
+  )
+}
+
+const ProfileCard = ({clickToProfile, showClose, username}: Props) => {
+  const userDetails = Container.useSelector(state => Tracker2Constants.getDetails(state, username))
+  const followThem = Container.useSelector(state => Tracker2Constants.followThem(state, username))
+  const followsYou = Container.useSelector(state => Tracker2Constants.followsYou(state, username))
+  const isSelf = Container.useSelector(state => state.config.username === username)
+
+  const dispatch = Container.useDispatch()
+  React.useEffect(() => {
+    dispatch(Tracker2Gen.createShowUser({asTracker: false, skipNav: true, username}))
```

Comment:
It seems like we're doing more work than we need to loading this every time. Could we only do this for some values of `userDetails.state`?

Reply:
This was intentional as I thought it made sense to refresh every time because thing can change since last time we loaded. But thinking about it more, I think it may not matter as much. Will will change it into only load when it's not available.